server/src/main/java/io/crate/planner/GCDangingArtifactsPlan.java

```diff
@@ -47,8 +48,10 @@ public void executeOrFail(DependencyCarrier dependencies,
             new OneRowActionListener<>(consumer, r -> r.isAcknowledged() ? new Row1(1L) : new Row1(0L));
 
         DeleteIndexRequest deleteRequest = new DeleteIndexRequest(IndexParts.DANGLING_INDICES_PREFIX_PATTERNS.toArray(new String[0]));
-        dependencies.transportActionProvider()
-            .transportDeleteIndexAction()
-            .execute(deleteRequest).whenComplete(listener);
+        dependencies.client().execute(
+            DeleteIndexAction.INSTANCE,
+            deleteRequest,
+            listener
+        );
```

Comment:
I was considering to remove the `listener` overload also eventually and only have `.execute(action, request) -> completableFuture`. That would require to prefer using `.execute(DeleteIndexAction.INSTANCE, request).whenComplete(listener)` here. See https://github.com/crate/crate/issues/11940

Not sure if we already need to do this now - could also be done later in a follow up I guess.

Reply:
I had a look but cannot find a quick resolution but to use `toBiConsumer`. Basically moving from `execute(...,ActionListener)` to `toBiConsumer`.

Reply:
`OneRowActionListener` already implements `BiConsumer`, you can change the code to

```
        dependencies.client().execute(DeleteIndexAction.INSTANCE, deleteRequest).whenComplete(listener);
```

Reply:
sorry did not see you reply. Then I could change them. Will look at all callers of `execute()`

Reply:
Could you review [143ffc6](https://github.com/crate/crate/pull/12343/commits/143ffc6db843ee3e7fc9c49c071d73c018843b7e) once the tests pass @mfussenegger? I tried to use `toBiComsumer()` only if necessary.

Reply:
For the listeners that don't implement BiConsumer it would have been fine to keep it as it is. But fine either way